Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <array>
+#include <optional>
+
+#include <QCheckBox>
+#include <QDialog>
+#include <QGroupBox>
+#include <QLineEdit>
+#include <QWidget>
+
+#include "Core/Core.h"
+#include "Core/IOS/USB/Emulated/Skylander.h"
+
+class QDialogButtonBox;
+class QLabel;
+class QPushButton;
+class QSpinBox;
+class QTabWidget;
+
+struct Skylander
+{
+  u8 portal_slot;
+  u16 sky_id;
+  u16 sky_var;
+};
+
+class SkylanderPortalWindow : public QWidget
+{
+  Q_OBJECT
+public:
+  explicit SkylanderPortalWindow(QWidget* parent = nullptr);
+  ~SkylanderPortalWindow() override;
+
+protected:
+  std::array<QLineEdit*, MAX_SKYLANDERS> edit_skylanders;
```

Comment:
```suggestion
  std::array<QLineEdit*, MAX_SKYLANDERS> m_edit_skylanders;
```
(sorry for missing this earlier)